core/hosting/HostWithCoreClrHost/src/SampleHost.cpp

```diff
@@ -0,0 +1,324 @@
+    if (last_slash != NULL)
+        *last_slash = 0;
+
+    // Construct the CoreCLR path
+    // For this sample, we know CoreCLR's path. For other hosts,
+    // it may be necessary to probe for coreclr.dll/libcoreclr.so
+    std::string coreClrPath(runtimePath);
+    coreClrPath.append(FS_SEPERATOR);
+    coreClrPath.append(CORECLR_FILE_NAME);
+
+    // Construct the managed library path
+    std::string managedLibraryPath(runtimePath);
+    managedLibraryPath.append(FS_SEPERATOR);
+    managedLibraryPath.append(MANAGED_ASSEMBLY);
+    
+    //
+    // STEP 1: Load CoreCLR (coreclr.dll/libcoreclr.so)
+    //
+#if WINDOWS
+    // <Snippet1>
+    HMODULE coreClr = LoadLibraryExA(coreClrPath.c_str(), NULL, 0);
+    // </Snippet1>
+#elif LINUX 
+    void *coreClr = dlopen(coreClrPath.c_str(), RTLD_NOW | RTLD_LOCAL);
+#endif 
+    if (coreClr == NULL)
+    {
+        printf("ERROR: Failed to load CoreCLR from %s\n", coreClrPath.c_str());
+        return -1;
+    }
+    else
+    {
+        printf("Loaded CoreCLR from %s\n", coreClrPath.c_str());
+    }
+
+    //
+    // STEP 2: Get CoreCLR hosting functions
+    //
+#if WINDOWS
+    // <Snippet2>
+    coreclr_initialize_ptr initializeCoreClr = (coreclr_initialize_ptr)GetProcAddress(coreClr, "coreclr_initialize");
+    coreclr_create_delegate_ptr createManagedDelegate = (coreclr_create_delegate_ptr)GetProcAddress(coreClr, "coreclr_create_delegate");
+    coreclr_shutdown_ptr shutdownCoreClr = (coreclr_shutdown_ptr)GetProcAddress(coreClr, "coreclr_shutdown");
+    // </Snippet2>
+#elif LINUX    
+    coreclr_initialize_ptr initializeCoreClr = (coreclr_initialize_ptr)dlsym(coreClr, "coreclr_initialize");
+    coreclr_create_delegate_ptr createManagedDelegate = (coreclr_create_delegate_ptr)dlsym(coreClr, "coreclr_create_delegate");
+    coreclr_shutdown_ptr shutdownCoreClr = (coreclr_shutdown_ptr)dlsym(coreClr, "coreclr_shutdown");
+#endif
+
+    if (initializeCoreClr == NULL)
+    {
+        printf("coreclr_initialize not found");
+        return -1;
+    }
+
+    if (createManagedDelegate == NULL)
+    {
+        printf("coreclr_create_delegate not found");
+        return -1;
+    }
+
+    if (shutdownCoreClr == NULL)
+    {
+        printf("coreclr_shutdown not found");
+        return -1;
+    }
+
+    //
+    // STEP 3: Construct AppDomain properties used when starting the runtime
+    //
+
+    // Construct the trusted platform assemblies (TPA) list
+    // This is the list of assemblies that .NET Core can load as
+    // trusted system assemblies (similar to the .NET Framework GAC).
+    // For this host (as with most), assemblies next to CoreCLR will 
+    // be included in the TPA list
+    std::string tpaList;
+    BuildTpaList(runtimePath, ".dll", tpaList);
+
+    // <Snippet3A>
+    // Define CoreCLR properties
+    const char* propertyKeys[] = {
+        "TRUSTED_PLATFORM_ASSEMBLIES",      // Trusted assemblies (like the GAC)
```

Comment:
I would remove any reference to the GAC when creating samples for CoreCLR.

Reply:
On line 121 above as well.

Reply:
I was using it as a way to help developers familiar with .NET Framework better understand what .NET Core's TPA list is. I can remove the references, though, if you think they're more confusing than helpful.

Reply:
Makes sense. I'll remove the GAC references.